src/js/components/landing/recentActivity/DeleteLink.jsx

```diff
@@ -30,56 +30,59 @@ export default class DeleteLink extends React.Component {
 	}
 
 	confirm(){
-		this.setState({active: !this.state.active});
+		this.setState({
+			active: true
+		})
 	}
 
 	canDelete(){
-		if(this.props.account.website_admin==true){
-			this.setState({delete: true});
-			return;
-		}else if(this.props.account.helpOnly==true){
-			this.setState({delete: false});
-			return;
+		let deletable = (this.props.account.website_admin || !this.props.account.helpOnly);
+
+		if(!deletable){
+			this.props.account.affiliations.forEach((aff)=>{
+				if(aff.agency_name == this.props.item.agency && (aff.permission == "writer" || aff.permission == 'submitter')){
+					deletable = true;
+					return;
+				}
+			})
 		}
-		this.props.account.affiliations.forEach((aff)=>{
-			if(aff.agency_name == this.props.item.agency && (aff.permission == "writer" || aff.permission == 'submitter')){
-				this.setState({delete:true});
-				return;
-			}
+
+		this.setState({
+			delete: deletable,
+			active: this.props.confirm
 		})
 	}
 
 	delete(){
-		ReviewHelper.deleteSubmission(this.props.submissionId)
-			.then((data) =>{
-				if(data.message == 'Success'){
-					this.confirm();
-					this.props.reload();
-				}else{
-					console.log('Delete Failed')
-				}
-			})
+		this.confirm();
+		this.props.reload();
+	}
+
+	reset(){
```

Comment:
Commenting here but really it'd be a code change farther up I think. There's no need to rebuild the rows when you hit cancel or the X. Is it a lot of work to update it so it only rebuilds when something actually changed? It's not a ton of processing time, but it's a principle thing.